### PR TITLE
fix(ingestion/tableau): human-readable message for PERMISSIONS_MODE_SWITCHED error

### PR DESCRIPTION
**Pull request by sid-acryl**
## Summary by CodeRabbit

- **Bug Fixes**
    - Improved error handling in Tableau connection object retrieval to provide clearer permission-related error messages.

- **Tests**
    - Added new tests for Tableau data source ingestion.
    - Introduced mock client setup for better testing of metadata, authentication, and permissions.

### CODE REVIEW

**Review by anshbansal — DISMISSED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/tableau.py [hunk 1]
```diff
@@ -882,10 +882,31 @@ def get_connection_object_page(
...
                     message=f"Received error fetching Query Connection {connection_type}",
-                    context=f"Errors: {errors}",
+                    context=f"Errors: {other_errors}",
                 )
+
+                if permission_mode_errors:
+                    self.report.warning(
+                        title="Derived Permission Error",
+                        message="Please check derived permissions as shared at https://community.tableau.com/s/question/0D54T00000QnjHbSAJ/how-to-fix-the-permissionsmodeswitched-error",
+                        context=f"${permission_mode_errors}",
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/tableau.py:907]:
```suggestion
                        context=f"{permission_mode_errors}",
```

File: metadata-ingestion/src/datahub/ingestion/source/tableau.py [hunk 2]
```diff
@@ -882,10 +882,31 @@ def get_connection_object_page(
                 error and (error.get(c.EXTENSIONS) or {}).get(c.SEVERITY) == c.WARNING
                 for error in errors
             ):
+                # filter out PERMISSIONS_MODE_SWITCHED to report error in human-readable format
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/tableau.py:885]: are the `PERMISSIONS_MODE_SWITCHED` errors marked as severity = warning?

File: metadata-ingestion/src/datahub/ingestion/source/tableau.py [hunk 3]
```diff
@@ -882,10 +882,31 @@ def get_connection_object_page(
...
+                        == "PERMISSIONS_MODE_SWITCHED"
+                    ):
+                        permission_mode_errors.append(error)
+                    else:
+                        other_errors.append(error)
+
                 self.report.warning(
                     message=f"Received error fetching Query Connection {connection_type}",
-                    context=f"Errors: {errors}",
+                    context=f"Errors: {other_errors}",
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/tableau.py:900]: add a `if other_errors` check

File: metadata-ingestion/src/datahub/ingestion/source/tableau.py [hunk 4]
```diff
@@ -882,10 +882,31 @@ def get_connection_object_page(
...
                 )
+
+                if permission_mode_errors:
+                    self.report.warning(
+                        title="Derived Permission Error",
+                        message="Please check derived permissions as shared at https://community.tableau.com/s/question/0D54T00000QnjHbSAJ/how-to-fix-the-permissionsmodeswitched-error",
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/tableau.py:906]: provide the actual suggestion right in here (e.g. "turn off X flag. See <link> for details.")

File: metadata-ingestion/src/datahub/ingestion/source/tableau.py (hunk 2 above)
Comment (sid-acryl) [reply metadata-ingestion/src/datahub/ingestion/source/tableau.py:1012]: Yes, below was the error message in report 

{'message': 'One or more of the attributes used in your filter contain sensitive data so your results have been automatically filtered to contain only the results you have permissions to see', 'extensions': {'severity': 'WARNING', 'code': 'PERMISSIONS_MODE_SWITCHED', 'properties': {'workbooksConnection': ['projectNameWithin']}}}

**Review by hsheth2 — APPROVED** [review]
can we add a test for this?

**Comment by sid-acryl** [comment]
> can we add a test for this?

Added a test case

**Pull request merged**